app/src/main/java/com/andrew/tasky/agenda/presentation/notifications/AgendaNotificationService.kt

```diff
@@ -0,0 +1,49 @@
+package com.andrew.tasky.agenda.presentation.notifications
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.navigation.NavDeepLinkBuilder
+import com.andrew.tasky.R
+import com.andrew.tasky.core.MainActivity
+
+class AgendaNotificationService(
+    private val context: Context
+) {
+
+    private val notificationManager = context
+        .getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+    fun showEventNotification(title: String, description: String) {
+        val eventNavIntent = NavDeepLinkBuilder(context)
+            .setGraph(R.navigation.my_nav)
+            .setDestination(R.id.reminderDetailFragment)
+            .createPendingIntent()
+
+        val intent = Intent(context, MainActivity::class.java)
+        intent.putExtra(EVENT_NAV_INTENT, eventNavIntent)
+
+        val pendingIntent = PendingIntent.getActivity(
+            context, 0, intent,
+            PendingIntent.FLAG_IMMUTABLE
+        )
+        val notification = NotificationCompat.Builder(context, EVENT_CHANNEL_ID)
+            .setSmallIcon(R.drawable.ic_notification_agenda)
+            .setContentTitle(title)
+            .setContentText(description)
+            .setContentIntent(pendingIntent)
+
+        notificationManager.notify(1, notification.build())
```

Comment:
If you call notify with the same ID again, it will replace an existing notification. Also, here it would never show 2+ notifications at once. The ID needs to be unique per notification (unless you want to update an existing one)